forc-plugins/forc-client/src/op/deploy.rs

```diff
@@ -251,48 +276,174 @@ async fn deploy_new_proxy(
     Ok(proxy_contract_id)
 }
 
-/// Builds and deploys contract(s). If the given path corresponds to a workspace, all deployable members
-/// will be built and deployed.
+/// Builds and deploys contracts, scripts, and predicates from the given path or workspace.
 ///
-/// Upon success, returns the ID of each deployed contract in order of deployment.
+/// Contracts are deployed directly, while scripts and predicates are deployed as blobs with generated loaders.
 ///
-/// When deploying a single contract, only that contract's ID is returned.
-pub async fn deploy(command: cmd::Deploy) -> Result<Vec<DeployedContract>> {
+/// Returns a vector of `DeployedPackage` representing all successful deployments.
+pub async fn deploy(command: cmd::Deploy) -> Result<Vec<DeployedPackage>> {
     if command.unsigned {
         println_warning("--unsigned flag is deprecated, please prefer using --default-signer. Assuming `--default-signer` is passed. This means your transaction will be signed by an account that is funded by fuel-core by default for testing purposes.");
     }
-
-    let mut deployed_contracts = Vec::new();
     let curr_dir = if let Some(ref path) = command.pkg.path {
         PathBuf::from(path)
     } else {
         std::env::current_dir()?
     };
-
-    let build_opts = build_opts_from_cmd(&command);
+    let build_opts = build_opts_from_cmd(&command, MemberFilter::default());
     let built_pkgs = built_pkgs(&curr_dir, &build_opts)?;
-    let pkgs_to_deploy = built_pkgs
+    let mut deployed_packages = Vec::new();
+
+    let contracts_to_deploy = built_pkgs
         .iter()
         .filter(|pkg| {
             pkg.descriptor
                 .manifest_file
                 .check_program_type(&[TreeType::Contract])
                 .is_ok()
         })
+        .cloned()
         .collect::<Vec<_>>();
 
-    if pkgs_to_deploy.is_empty() {
-        println_warning("No deployable contracts found in the current directory.");
+    let scripts_to_deploy = built_pkgs
+        .iter()
+        .filter(|pkg| {
+            pkg.descriptor
+                .manifest_file
+                .check_program_type(&[TreeType::Script])
+                .is_ok()
+        })
+        .cloned()
+        .collect::<Vec<_>>();
+
+    let predicates_to_deploy = built_pkgs
+        .iter()
+        .filter(|pkg| {
+            pkg.descriptor
+                .manifest_file
+                .check_program_type(&[TreeType::Predicate])
+                .is_ok()
+        })
+        .cloned()
+        .collect::<Vec<_>>();
+
+    if contracts_to_deploy.is_empty()
+        && scripts_to_deploy.is_empty()
+        && predicates_to_deploy.is_empty()
+    {
+        println_warning("No deployable package is found in the current directory.");
```

Comment:
nit

```suggestion
        println_warning("No deployable package was found in the current directory.");
```

Reply:
Thanks! I am waiting to see if other I'll get more stuff to get them all at once.